File: Interfaces/IGrHeaderService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface IGrHeaderService
    {
        Task<ApiResponse<IEnumerable<GrHeaderDto>>> GetAllAsync();
        Task<ApiResponse<GrHeaderDto?>> GetByIdAsync(int id);
        Task<ApiResponse<GrHeaderDto?>> GetByERPDocumentNoAsync(string erpDocumentNo);
        Task<ApiResponse<GrHeaderDto>> CreateAsync(CreateGrHeaderDto createDto);
        Task<ApiResponse<GrHeaderDto>> UpdateAsync(int id, UpdateGrHeaderDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(int id);
        Task<ApiResponse<IEnumerable<GrHeaderDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<GrHeaderDto>>> GetByBranchCodeAsync(string branchCode);
        Task<ApiResponse<IEnumerable<GrHeaderDto>>> GetByCustomerCodeAsync(string customerCode);
        Task<ApiResponse<IEnumerable<GrHeaderDto>>> GetByDateRangeAsync(DateTime startDate, DateTime endDate);
    }
}

File: WebApi/Interfaces/IGrHeaderService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { GrHeaderDto, CreateGrHeaderDto, UpdateGrHeaderDto } from '../Models/GrHeaderDtos';

export interface IGrHeaderService {
  getAll(): Promise<ApiResponse<GrHeaderDto[]>>;
  getById(id: number): Promise<ApiResponse<GrHeaderDto | null>>;
  getByERPDocumentNo(erpDocumentNo: string): Promise<ApiResponse<GrHeaderDto | null>>;
  create(createDto: CreateGrHeaderDto): Promise<ApiResponse<GrHeaderDto>>;
  update(id: number, updateDto: UpdateGrHeaderDto): Promise<ApiResponse<GrHeaderDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
  getActive(): Promise<ApiResponse<GrHeaderDto[]>>;
  getByBranchCode(branchCode: string): Promise<ApiResponse<GrHeaderDto[]>>;
  getByCustomerCode(customerCode: string): Promise<ApiResponse<GrHeaderDto[]>>;
  getByDateRange(startDate: Date, endDate: Date): Promise<ApiResponse<GrHeaderDto[]>>;
}